dawn/src/dawn/Optimizer/OptimizerContext.cpp

```diff
@@ -599,7 +599,20 @@ OptimizerContext::OptimizerContext(DiagnosticsEngine& diagnostics, OptimizerCont
                                    const std::shared_ptr<SIR>& SIR)
     : diagnostics_(diagnostics), options_(options), SIR_(SIR) {
   DAWN_LOG(INFO) << "Intializing OptimizerContext ... ";
+  if(SIR)
+    fillIIR();
 }
+
+OptimizerContext::OptimizerContext(
+    DiagnosticsEngine& diagnostics, OptimizerContextOptions options,
+    std::map<std::string, std::shared_ptr<iir::StencilInstantiation>> const&
+        stencilInstantiationMap)
+    : diagnostics_(diagnostics), options_(options), SIR_() {
+  DAWN_LOG(INFO) << "Intializing OptimizerContext from stencil instantiation map ... ";
+  for(auto& [name, stencilInstantiation] : stencilInstantiationMap)
+    restoreIIR(name, stencilInstantiation);
```

Comment:
if we call restore here, the optimizer will have 2 passes in the pass-list already. These should probably be cleaned up again, right?

See 
https://github.com/MeteoSwiss-APN/dawn/blob/98c6c0c319494cd72bf901363b727da37718eed1/dawn/src/dawn/Optimizer/OptimizerContext.cpp#L735-L736

Reply:
In this case we don't re-use that OptimizerContext for the further optimization passes so it's valid not to clean them up, but I added the clean-up anyway because that could change in the future.